tests/Unit/OutputFormatterTest.php

```diff
@@ -0,0 +1,300 @@
+    }
+
+    /**
+     * @test
+     */
+    public function implodeWithOneStringValueReturnsStringValue(): void
+    {
+        $value = 'tea';
+        $values = [$value];
+
+        $result = $this->subject->implode(', ', $values);
+
+        self::assertSame($value, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function implodeWithMultipleStringValuesReturnsValuesSeparatedBySeparator(): void
+    {
+        $value1 = 'tea';
+        $value2 = 'coffee';
+        $values = [$value1, $value2];
+        $separator = ', ';
+
+        $result = $this->subject->implode(', ', $values);
+
+        self::assertSame($value1 . $separator . $value2, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function implodeWithOneRenderableReturnsRenderedRenderable(): void
+    {
+        $renderable = $this->createMock(Renderable::class);
+        $renderedRenderable = 'tea';
+        $renderable->method('render')->with($this->outputFormat)->willReturn($renderedRenderable);
+        $values = [$renderable];
+
+        $result = $this->subject->implode(', ', $values);
+
+        self::assertSame($renderedRenderable, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function implodeWithMultipleRenderablesReturnsRenderedRenderablesSeparatedBySeparator(): void
+    {
+        $renderable1 = $this->createMock(Renderable::class);
+        $renderedRenderable1 = 'tea';
+        $renderable1->method('render')->with($this->outputFormat)->willReturn($renderedRenderable1);
+        $renderable2 = $this->createMock(Renderable::class);
+        $renderedRenderable2 = 'tea';
+        $renderable2->method('render')->with($this->outputFormat)->willReturn($renderedRenderable2);
+        $values = [$renderable1, $renderable2];
+        $separator = ', ';
+
+        $result = $this->subject->implode($separator, $values);
+
+        self::assertSame($renderedRenderable1 . $separator . $renderedRenderable2, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function implodeWithIncreaseLevelFalseUsesDefaultIndentationLevelForRendering(): void
+    {
+        $renderable = $this->createMock(Renderable::class);
+        $renderedRenderable = 'tea';
+        $renderable->method('render')->with($this->outputFormat)->willReturn($renderedRenderable);
+        $values = [$renderable];
+
+        $result = $this->subject->implode(', ', $values, false);
+
+        self::assertSame($renderedRenderable, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function implodeWithIncreaseLevelTrueIncreasesIndentationLevelForRendering(): void
+    {
+        $renderable = $this->createMock(Renderable::class);
+        $renderedRenderable = 'tea';
+        $renderable->method('render')->with($this->outputFormat->nextLevel())->willReturn($renderedRenderable);
+        $values = [$renderable];
+
+        $result = $this->subject->implode(', ', $values, true);
+
+        self::assertSame($renderedRenderable, $result);
+    }
```

Comment:
I don't see the level increment being asserted here.  Shouldn't we be getting tea after some space?

Reply:
I missed the change to the argument passed to `with()` compared with the previous test.  I think I see and understand what's going on now.  Reviewing this has been an educative process 👍